src/kpi/statistics/issueTracking/issueTracking.service.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { RepositoryIdentifier } from '../../../entities/repositories/model';
+import { RepositoryDocument } from '../../../entities/repositories/model/schemas';
+import {
+  calculateAvgCapability,
+  calculateAvgEfficiency,
+  calculateAvgRate,
+  mapReleasesToIssues,
+} from './lib';
+import { transformMapToObject } from '../lib';
+import { RepositoryService } from '../../../entities/repositories/repository.service';
+
+@Injectable()
+export class IssueTrackingService {
+  private readonly logger = new Logger(IssueTrackingService.name);
+
+  constructor(private readonly repoService: RepositoryService) {}
+
+  async issueCompletionRate(
+    repoIdent: RepositoryIdentifier,
+    labelNames?: string[],
+  ) {
+    const repo = await this.repoService.read(
+      { ...repoIdent },
+      { commits: false, diffs: false },
+    );
+    await this.populateRepository(repo, labelNames);
+
+    const releaseIssueMap = mapReleasesToIssues(repo.releases, repo.issues);
+    const { avgRate, rateMap } = calculateAvgRate(releaseIssueMap);
+    return {
+      avgRate: avgRate,
+      rawData: transformMapToObject(rateMap),
+    };
+  }
+
+  async issueCompletionCapability(
+    repoIdent: RepositoryIdentifier,
+    labelNames?: string[],
+    timeToComplete: number = 14 * 24 * 60 * 60 * 1000,
+  ) {
+    const repo = await this.repoService.read(
+      { ...repoIdent },
+      { commits: false, diffs: false },
+    );
+    await this.populateRepository(repo, labelNames);
+
+    const releaseIssueMap = mapReleasesToIssues(repo.releases, repo.issues);
+    const { capabilityMap, avgCapability } = calculateAvgCapability(
+      releaseIssueMap,
+      timeToComplete,
+    );
+
+    return { avgCapability, rawData: transformMapToObject(capabilityMap) };
+  }
+
+  async issueCompletionEfficiency(
+    repoIdent: RepositoryIdentifier,
+    labelNames?: string[],
+    timeToComplete: number = 14 * 24 * 60 * 60 * 1000,
+  ) {
+    const repo = await this.repoService.read(
+      { ...repoIdent },
+      { commits: false, diffs: false },
+    );
+    await this.populateRepository(repo, labelNames);
+
+    const releaseIssueMap = mapReleasesToIssues(repo.releases, repo.issues);
+    const { efficiencyMap, avgEfficiency } = calculateAvgEfficiency(
+      releaseIssueMap,
+      timeToComplete,
+    );
+
+    return { avgEfficiency, rawData: transformMapToObject(efficiencyMap) };
+  }
+
+  private async populateRepository(
+    repo: RepositoryDocument,
+    labelNames?: string[],
+  ) {
+    await repo.populate({
+      path: 'releases',
+      options: {
+        sort: 'created_at',
+      },
+    });
+    await repo.populate({
```

Comment:
I'm pretty sure we can do repo.populate({releases}).populate({issues}) and do this in one query.

from the mongoose documentation
![image](https://user-images.githubusercontent.com/8062521/154048946-ce28a4c1-c15e-400c-bc1e-6c88b8c0f25f.png)